tests/unit/modules/network/iosxr/test_iosxr_snmp_server.py

```diff
@@ -0,0 +1,2488 @@
+                "subscriber": {
+                    "session_agg_node": True,
+                    "session_agg_access_interface": True,
+                },
+                "addrpool": {"low": True, "high": True},
+                "cisco_entity_ext": True,
+                "entity_state": {"operstatus": True, "switchover": True},
+                "entity_redundancy": {
+                    "all": True,
+                    "status": True,
+                    "switchover": True,
+                },
+            },
+            "chassis_id": "test2",
+            "contact": "test",
+            "location": "test",
+            "targets": [
+                {"name": "test1", "vrf": "vrf1"},
+                {"name": "test1", "host": "1.1.1.1"},
+            ],
+            "context": ["test", "test2"],
+            "logging_threshold_oid_processing": 0,
+            "logging_threshold_pdu_processing": 0,
+            "mib_bulkstat_max_procmem_size": 100,
+            "mib_object_lists": ["test"],
+            "mib_bulkstat_transfer_ids": [
+                {
+                    "name": "test1",
+                    "retry": 1,
+                    "buffer_size": 1024,
+                    "enable": True,
+                    "format_schemaASCI": True,
+                    "retain": 1,
+                    "schema": "test",
+                }
+            ],
+            "timeouts": {
+                "duplicate": 0,
+                "inQdrop": 0,
+                "subagent": 1,
+                "pdu_stats": 1,
+                "threshold": 0,
+            },
+            "packetsize": 485,
+            "correlator": {"buffer_size": 1024},
+            "trap_source": "GigabitEthernet0/0/0/1",
+            "throttle_time": 50,
+            "community_map": [
+                {"name": "test", "context": "test", "security_name": "test2"},
+                {"name": "test1", "context": "test", "security_name": "test2"},
+            ],
+            "inform": {"pending": 1, "retries": 1, "timeout": 1},
+            "oid_poll_stats": True,
+            "overload_control": {
+                "overload_drop_time": 0,
+                "overload_throttle_rate": 0,
+            },
+            "interfaces": [
+                {
+                    "name": "GigabitEthernet0/0/0/0",
+                    "notification_linkupdown_disable": True,
+                }
+            ],
+            "ifmib": {
+                "ifalias_long": True,
+                "internal_cache_max_duration": 0,
+                "ipsubscriber": True,
+                "stats": True,
+            },
+            "ifindex": True,
+            "mroutemib_send_all_vrf": True,
+            "notification_log_mib": {
+                "size": 1,
+                "default": True,
+                "disable": True,
+                "GlobalSize": 1,
+            },
+        }
+        print(result["parsed"])
+        self.assertEqual(parsed_list, result["parsed"])
```

Comment:
Can you add testcase for state: overridden and to check the idempotent behavior for various states? Any specific reason to have the print statements in these tests?

Reply:
ack